File: api/state.py
from http.server import BaseHTTPRequestHandler

from api._lib.auth import require_user
from api._lib.http import method_not_allowed, read_json_body, send_empty, send_error, send_json
from api._lib.store import get_workspace_state, save_workspace_state


class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        try:
            token, user = require_user(self.headers.get("Authorization"))
            state = get_workspace_state(token, user["id"])
            send_json(self, 200, state)
        except Exception as exc:
            send_error(self, exc)

    def do_POST(self):
        try:
            token, user = require_user(self.headers.get("Authorization"))
            state = save_workspace_state(token, user["id"], read_json_body(self))
            send_json(self, 200, {"success": True, **state})
        except Exception as exc:
            send_error(self, exc)

    def do_OPTIONS(self):
        send_empty(self)

    def do_DELETE(self):
        method_not_allowed(self, "GET, POST")

File: api/state.js
import { requireUser } from "./_lib/auth.js";
import { getWorkspaceState, saveWorkspaceState } from "./_lib/store.js";

export default async function handler(req, res) {
  try {
    const { token, user } = await requireUser(req);
    
    if (req.method === "GET") {
      const state = await getWorkspaceState(token, user.id);
      return res.status(200).json(state);
    }
    
    if (req.method === "POST") {
      const state = await saveWorkspaceState(token, user.id, req.body || {});
      return res.status(200).json({ success: true, ...state });
    }
    
    res.status(405).json({ error: "Method not allowed" });
  } catch (err) {
    res.status(err.status || 500).json({ error: err.message });
  }
}